### PR TITLE
Add streaming chat completions

### PR DESCRIPTION
**Pull request by bakks**
Hey all - I'm just piggybacking on https://github.com/PullRequestInc/go-gpt3/pull/30 and adding streaming support. Recommend that you merge @abatilo 's code first so that my additions are more clear. And please feel free to edit this PR yourself.

### CODE REVIEW

**Review by appleboy — APPROVED** [review]

File: gpt3.go
Comment (bakks) [reply gpt3.go:226]: I believe this line is here partly to filter out empty lines in the streaming response, of which there are many, so I don't think it makes sense to add extra logging here.

Some background context - the OpenAI documentation on the streaming flag is:
> Whether to stream back partial progress. If set, tokens will be sent as data-only [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#Event_stream_format) as they become available, with the stream terminated by a `data: [DONE]` message.

What I saw when debugging the responses is that every bit of data indeed had the `data:` prefix, except for the completely empty lines. So this line is exercised during normal operation, and the JSON unmarshalling will take care of throwing an error if there's line corruption.

**Comment by bakks** [comment]
Please merge when ready, i don't have write access

**Comment by bakks** [comment]
I've rebased and pushed after the merge of https://github.com/PullRequestInc/go-gpt3/pull/30, so this branch should only show my own changes now. Merge when ready.

**Review by tylermann — APPROVED** [review]
Thanks for the streaming support for this one too. I'll go ahead and merge this in.

**Pull request merged**
